### PR TITLE
Animation Tracks: Convert to es6 classes

### PR DESCRIPTION
**Pull request by ianpurvis**
Supports #19986

### CODE REVIEW

File: src/animation/KeyframeTrack.js [hunk 1]
```diff
@@ -467,6 +453,16 @@ Object.assign( KeyframeTrack.prototype, {
 
 	}
 
+}
+
+Object.assign( KeyframeTrack.prototype, {
+
+	TimeBufferType: Float32Array,
```
Comment (ianpurvis) [inline src/animation/KeyframeTrack.js:460]: I debated whether or not to preserve these prototype properties. They should reduce memory footprint with a large number of animation tracks, though I don't know if that matters in real world usage. They seem better suited as static props than instance props? Anyways, let me know what direction is good and I can tweak this PR as needed.

Honestly, all of the inheritance doesn't seem that useful. In the long run maybe the track system could just be refactored with a more data-driven approach.

File: src/animation/tracks/QuaternionKeyframeTrack.js [hunk 2]
```diff
@@ -5,29 +5,24 @@ import { QuaternionLinearInterpolant } from '../../math/interpolants/QuaternionL
...
 
 	DefaultInterpolation: InterpolateLinear,
 
-	InterpolantFactoryMethodLinear: function ( result ) {
-
-		return new QuaternionLinearInterpolant( this.times, this.values, this.getValueSize(), result );
-
-	},
-
 	InterpolantFactoryMethodSmooth: undefined // not yet implemented
```
Comment (ianpurvis) [inline src/animation/tracks/QuaternionKeyframeTrack.js:26]: I noticed this pattern where prototype methods are undefined to drive fallback logic in base `KeyframeTrack`. It feels kind of wrong to mutate the KeyframeTrack interface, although I guess if these can be considered internal it doesn't matter. No need to change now, just pointing it out as another candidate for some cleanup down the road.

File: src/animation/KeyframeTrack.js [hunk 3]
```diff
@@ -467,6 +453,16 @@ Object.assign( KeyframeTrack.prototype, {
...
+
+	ValueBufferType: Float32Array,
+
+	DefaultInterpolation: InterpolateLinear,
+
 } );
```
Comment (ianpurvis) [inline src/animation/KeyframeTrack.js:466]: I debated whether or not to preserve these prototype properties. They should reduce memory footprint with a large number of animation tracks, though I don't know if that matters in real world usage. They seem better suited as static props than instance props? Anyways, let me know what direction is good and I can tweak this PR as needed.

Honestly, all of the inheritance doesn't seem that useful. In the long run maybe the track system could just be refactored with a more data-driven approach.

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
Btw, seems like `Object.assign()` breaks tree-shaking in Rollup. I'll change them to individual `*.prototype.* =`.

**Comment by mrdoob** [comment]
https://github.com/mrdoob/three.js/commit/6100437b3e4d83c176e8ec82bf764e71df32f308

**Comment by Mugen87** [comment]
It seems this PR has broken keyframe serialization. This can easily by drag'n'drop an animated asset into the editor and wait for the save.

https://raw.githack.com/mrdoob/three.js/dev/editor/index.html
https://github.com/mrdoob/three.js/blob/dev/examples/models/gltf/Flamingo.glb

The following runtime error occurs:

> three.module.js:32997 Uncaught RangeError: Maximum call stack size exceeded

**Comment by Mugen87** [comment]
Fiddle demonstrating the bug:

https://jsfiddle.net/th1m48a9/

**Comment by ianpurvis** [comment]
@Mugen87 Thanks for the bug report... I think this should be fixed in #21342.

**Pull request merged**